addons/mail/controllers/mail.py

```diff
@@ -132,6 +144,8 @@ def _redirect_to_record(cls, model, res_id, access_token=None, **kwargs):
         view_id = record_sudo.get_formview_id()
         if view_id:
             url_params['view_id'] = view_id
+        if highlight_message_id := kwargs.get("highlight_message_id"):
+            url_params["highlight_message_id"] = highlight_message_id
```

Comment:
I'm not sure but maybe create a function for this because this is repeated in many places.